### ISSUE TITLE
Should I be root (or use sudo) when creating recipes?

### ISSUE THREAD

**Issue by cinnamonmatexfce** [AppImageCrafters/appimage-builder#221]
ArchLinux, I'm using this path:
`/home/username/.local/bin/appImage-builder/AppDir/usr/bin/PROGRAM`

I'm executing `appimage-builder` from its directory (see above):
`$ ./appimage-builder-1.0.0-d3ce41f-x86_64.AppImage --generate`
```
INFO:FilePackageResolver:/usr/bin/pacman -Fy /usr/lib/libSDL2-2.0.so.0 /usr/lib/libGL.so.1 /usr/lib/libGLU.so.1 /usr/lib/libz.so.1 /usr/lib/libm.so.6 /usr/lib/libstdc++.so.6 /usr/lib/libSDL2_net-2.0.so.0 /usr/lib/libgcc_s.so.1 /usr/lib/libc.so.6 /usr/lib/libGLdispatch.so.0 /usr/lib/libGLX.so.0 /usr/lib/libOpenGL.so.0 /usr/lib/libpthread.so.0 /usr/lib/libdl.so.2 /usr/lib/libX11.so.6 /usr/lib/libxcb.so.1 /usr/lib/libXau.so.6 /usr/lib/libXdmcp.so.6 /lib64/ld-linux-x86-64.so.2 /usr/lib/ld-linux-x86-64.so.2
error: you cannot perform this operation unless you are root.
```

P. S.: `AppImageBuilder.yml` has been created anyway...

**Comment by azubieta** [comment]
No, at least your application requires it. This is likely a bug.

I'm not a Archlinux expert so if you find a way to run the command from the logs without root please let me know so I can correct it.

**Comment by cinnamonmatexfce** [comment]
> No, at least your application requires it. This is likely a bug.
> 
> I'm not a Archlinux expert so if you find a way to run the command from the logs without root please let me know so I can correct it.

I didn't try actually yet... But... I had a look closely to the output and indeed `/usr/bin/pacman -Fy`requires elevated privileges, precisely the `y` switch:

> -y, --refresh
> 
>Download a fresh copy of the master package >database from the server(s) defined in >pacman.conf(5). This should typically be used >each time you use --sysupgrade or -u. Passing >two --refresh or -y flags will force a refresh of >all package databases, even if they appear to >be up-to-date.

**Comment by azubieta** [comment]
So the solution would be to remove the `-y` ?
It would be great If you could check the `pacman.conf` at `appimage-build/pacman` we use and provide an alternative setup that allows using a cache different from the one in the system (that's how we do it apt).
